vcd/resource_vcd_vapp.go

```diff
@@ -33,81 +41,112 @@ func resourceVcdVApp() *schema.Resource {
 			},
 			"vdc": {
 				Type:        schema.TypeString,
-				Required:    false,
 				Optional:    true,
 				ForceNew:    true,
 				Description: "The name of VDC to use, optional if defined at provider level",
 			},
 			"template_name": {
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of the vApp Template to use",
+				Deprecated:  implicitVmInVappDeprecation,
 			},
 			"catalog_name": {
-				Type:     schema.TypeString,
-				Optional: true,
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The catalog name in which to find the given vApp Template",
+				Deprecated:  implicitVmInVappDeprecation,
 			},
 			"network_name": {
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Name of the network this vApp should join",
+				Deprecated:  implicitVmInVappDeprecation,
```

Comment:
Why is network being deprecated? Please see in vCD API docs and/or UI if you can connect/"assign" an empty vApp to a network without a VM.

CC: @Didainius @vbauzysvmware

Reply:
I look at this description and I think it has a bit incorrect comment. In fact this is the `network_name` to which the single VM inside vApp (old behavior) will be attached (the org network will be attached to vApp as a side effect)

Currently our `vcd_vapp_vm` resources automa**g**ically attach a network to vApp if it is not attached, but in future we should fix it with https://github.com/terraform-providers/terraform-provider-vcd/issues/264 (a breaking change once we remove that *automagic* part of attachment from vcd_vapp_vm code).
Field `network_name` wouldn't still suit it because it must accept a set of networks, not a single one. And for now I am leaning towards using IDs instead of names (this field being `org_network_ids` rather than `org_networks`). Especially that our networks will have proper IDs after Giuseppe's merge a few days ago.

Reply:
Not sure I agree with the use of IDs, just because it's harder for user, but otherwise good comment.

> I look at this description and I think it has a bit incorrect description. In fact this is the network_name to which the single VM inside vApp (old behavior) will be attached (the org network will be attached to vApp as a side effect)

I would like the current misleading description to be updated to the correct one. Your write-up above is almost a perfect fit for the new description. Even if it takes so many words to explain, we need to make it clear what the field does and that we are deprecating it for a reason. Also, description should be clarified in the doc page too. @dataclouder please look into this.

Reply:
We are deprecating everything that goes to the implicit VM. The network is one of such properties.
Currently, we don't assign networks to the vApp directly. We assign them to the VM, and from there it goes assigned to the vApp. We should change this behavior, but at the moment the vApp resource doesn't own the network.

Reply:
For the record, I copied the descriptions from the vApp documentation page (docs/r/vapp.html.markdown). If that is not correct, we should change both.